ruleset-style.md

```diff
@@ -0,0 +1,90 @@
+# Ruleset style guide
+
+Goal: Rules should be written in a way that is consistent, easy for humans to
+read and debug, and reduces the chance of errors, and makes testing easy.
+
+To that end, here are some style guidelines for writing or modifying rulesets.
+They are intended to help and simplify in places where choices are ambiguous,
+but like all guidelines they can be broken if the circumstances require it.
+
+Avoid using the left-wildcard ("&lt;target host='*.example.com'&gt;") unless you
+really mean it. Many rules today specify a left-wildcard target, but the
+rewrite rules only rewrite an explicit list of hostnames.
+
+Instead, prefer listing explicit target hosts and a single rewrite from "^http:" to
+"^https:". This saves you time as a ruleset author because each explicit target
+host automatically creates a an implicit test URL, reducing the need to add your
+own test URLs. These also make it easier for someone reading the ruleset to figure out
+which subdomains are covered.
+
+If you know all subdomains of a given domain support HTTPS, go ahead and use a
+left-wildcard, along with a plain rewrite from "^http:" to "^https:". Make sure
+to add a bunch of test URLs for the more important subdomains. If you're not
+sure what subdomains might exist, check the 'subdomain' tab on Wolfram Alpha:
+http://www.wolframalpha.com/input/?i=_YOUR_DOMAIN_GOES_HERE_.
+
+If there are a handful of tricky subdomains, but most subdomains can handle the
+plain rewrite from "^http:" to "^https:", specify the rules for the tricky
+subdomains first, and then then plain rule last. Earlier rules will take
+precedence, and processing stops at the first matching rule.
+
```

Comment:
If this is the case, wouldn't you want the most common subdomains at the top for the most efficient rule processing?